spec/acceptance/docker_full_spec.rb

```diff
@@ -0,0 +1,643 @@
+
+        docker::run { 'container_2_4_2':
+          image   => 'newos',
+          command => 'init',
+          use_name => true,
+          require => Docker::Image['newos'],
+        }
+      EOS
+
+      apply_manifest(pp, :catch_failures => true)
+      unless fact('selinux') == 'true'
+        apply_manifest(pp, :catch_changes => true)
+      end
+
+      # Commit currently running container as an image called newos
+      container_id = shell("docker ps | awk 'FNR == 2 {print $1}'")
+      shell("docker commit #{container_id.stdout.strip} newos")
+
+      # Stop all container using systemd
+      shell('ls -D -1 /etc/systemd/system/docker-container* | sed \'s/\/etc\/systemd\/system\///g\' | sed \'s/\.service//g\' | while read container; do service $container stop; done')
+
+      # Stop all running containers
+      shell('docker rm -f $(docker ps -a -q) || true')
+
+      # Make sure no other containers are running
+      shell('docker ps | wc -l') do |r|
+        expect(r.stdout).to match(/^1$/)
+      end
+
+      # Export new to a tar file
+      shell("docker save newos > /root/rootfs.tar")
+
+      # Remove all images
+      shell('docker rmi $(docker images -q) || true')
+
+      # Make sure no other images are present
+      shell('docker images | wc -l') do |r|
+        expect(r.stdout).to match(/^1$/)
+      end
+
+      apply_manifest(pp2, :catch_failures => true)
+      unless fact('selinux') == 'true'
+        apply_manifest(pp2, :catch_changes => true)
+      end
+
+      container_id = shell("docker ps | awk 'FNR == 2 {print $1}'")
+      shell("docker exec #{container_id.stdout.strip} ls /root") do |r|
+        expect(r.stdout).to match(/test_file_for_tar_test.txt/)
+      end
+    end
+
+    after(:each) do
+      # Stop all container using systemd
+      shell('ls -D -1 /etc/systemd/system/docker-container* | sed \'s/\/etc\/systemd\/system\///g\' | sed \'s/\.service//g\' | while read container; do service $container stop; done')
+      # Delete all running containers
+      shell('docker rm -f $(docker ps -a -q) || true')
+      # Check to make sure no running containers are present
+      shell('docker ps | wc -l') do |r|
+        expect(r.stdout).to match(/^0|1$/)
+      end
+    end
+  end
+
+  context 'test 2.5 - Delete Image' do
+    it 'runs test' do
+      pp=<<-EOS
+        class { 'docker':}
+        docker::image { 'busybox':
+          ensure => present,
+          require => Class['docker'],
+        }
+      EOS
+
+      pp2=<<-EOS
+        class { 'docker':}
+        docker::image { 'busybox':
+          ensure => absent,
+        }
+      EOS
+
+      apply_manifest(pp, :catch_failures => true)
+      apply_manifest(pp2, :catch_failures => true)
+      unless fact('selinux') == 'true'
+        apply_manifest(pp2, :catch_changes => true)
+      end
+
+      shell('docker images') do |r|
+        expect(r.stdout).to_not match(/ubuntu/)
+      end
+    end
+  end
+end
+
+[true,false].each do |use_name|
```

Comment:
As of https://github.com/garethr/garethr-docker/pull/222 I don't think this is now required.